File: bindings/lua/Polycode/TimeInfo.lua
class "TimeInfo"


function TimeInfo:__getvar(name)
	if name == "seconds" then
		return Polycode.TimeInfo_get_seconds(self.__ptr)
	elseif name == "minutes" then
		return Polycode.TimeInfo_get_minutes(self.__ptr)
	elseif name == "hours" then
		return Polycode.TimeInfo_get_hours(self.__ptr)
	elseif name == "month" then
		return Polycode.TimeInfo_get_month(self.__ptr)
	elseif name == "monthDay" then
		return Polycode.TimeInfo_get_monthDay(self.__ptr)
	elseif name == "weekDay" then
		return Polycode.TimeInfo_get_weekDay(self.__ptr)
	elseif name == "year" then
		return Polycode.TimeInfo_get_year(self.__ptr)
	elseif name == "yearDay" then
		return Polycode.TimeInfo_get_yearDay(self.__ptr)
	end
end

function TimeInfo:__setvar(name,value)
	if name == "seconds" then
		Polycode.TimeInfo_set_seconds(self.__ptr, value)
		return true
	elseif name == "minutes" then
		Polycode.TimeInfo_set_minutes(self.__ptr, value)
		return true
	elseif name == "hours" then
		Polycode.TimeInfo_set_hours(self.__ptr, value)
		return true
	elseif name == "month" then
		Polycode.TimeInfo_set_month(self.__ptr, value)
		return true
	elseif name == "monthDay" then
		Polycode.TimeInfo_set_monthDay(self.__ptr, value)
		return true
	elseif name == "weekDay" then
		Polycode.TimeInfo_set_weekDay(self.__ptr, value)
		return true
	elseif name == "year" then
		Polycode.TimeInfo_set_year(self.__ptr, value)
		return true
	elseif name == "yearDay" then
		Polycode.TimeInfo_set_yearDay(self.__ptr, value)
		return true
	end
	return false
end
function TimeInfo:TimeInfo(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.TimeInfo(unpack(arg))
	end
end

function TimeInfo:__delete()
	if self then Polycode.delete_TimeInfo(self.__ptr) end
end

File: bindings/javascript/Polycode/TimeInfo.js
function TimeInfo() {
	Object.defineProperties(this, {
		'seconds': { enumerable: true, configurable: true, get: TimeInfo.prototype.__get_seconds, set: TimeInfo.prototype.__set_seconds},
		'minutes': { enumerable: true, configurable: true, get: TimeInfo.prototype.__get_minutes, set: TimeInfo.prototype.__set_minutes},
		'hours': { enumerable: true, configurable: true, get: TimeInfo.prototype.__get_hours, set: TimeInfo.prototype.__set_hours},
		'month': { enumerable: true, configurable: true, get: TimeInfo.prototype.__get_month, set: TimeInfo.prototype.__set_month},
		'monthDay': { enumerable: true, configurable: true, get: TimeInfo.prototype.__get_monthDay, set: TimeInfo.prototype.__set_monthDay},
		'weekDay': { enumerable: true, configurable: true, get: TimeInfo.prototype.__get_weekDay, set: TimeInfo.prototype.__set_weekDay},
		'year': { enumerable: true, configurable: true, get: TimeInfo.prototype.__get_year, set: TimeInfo.prototype.__set_year},
		'yearDay': { enumerable: true, configurable: true, get: TimeInfo.prototype.__get_yearDay, set: TimeInfo.prototype.__set_yearDay}
	})
}
TimeInfo.prototype.__get_seconds = function() {
	return Polycode.TimeInfo__get_seconds(this.__ptr)
}

TimeInfo.prototype.__set_seconds = function(val) {
	Polycode.TimeInfo__set_seconds(this.__ptr, val)
}

TimeInfo.prototype.__get_minutes = function() {
	return Polycode.TimeInfo__get_minutes(this.__ptr)
}

TimeInfo.prototype.__set_minutes = function(val) {
	Polycode.TimeInfo__set_minutes(this.__ptr, val)
}

TimeInfo.prototype.__get_hours = function() {
	return Polycode.TimeInfo__get_hours(this.__ptr)
}

TimeInfo.prototype.__set_hours = function(val) {
	Polycode.TimeInfo__set_hours(this.__ptr, val)
}

TimeInfo.prototype.__get_month = function() {
	return Polycode.TimeInfo__get_month(this.__ptr)
}

TimeInfo.prototype.__set_month = function(val) {
	Polycode.TimeInfo__set_month(this.__ptr, val)
}

TimeInfo.prototype.__get_monthDay = function() {
	return Polycode.TimeInfo__get_monthDay(this.__ptr)
}

TimeInfo.prototype.__set_monthDay = function(val) {
	Polycode.TimeInfo__set_monthDay(this.__ptr, val)
}

TimeInfo.prototype.__get_weekDay = function() {
	return Polycode.TimeInfo__get_weekDay(this.__ptr)
}

TimeInfo.prototype.__set_weekDay = function(val) {
	Polycode.TimeInfo__set_weekDay(this.__ptr, val)
}

TimeInfo.prototype.__get_year = function() {
	return Polycode.TimeInfo__get_year(this.__ptr)
}

TimeInfo.prototype.__set_year = function(val) {
	Polycode.TimeInfo__set_year(this.__ptr, val)
}

TimeInfo.prototype.__get_yearDay = function() {
	return Polycode.TimeInfo__get_yearDay(this.__ptr)
}

TimeInfo.prototype.__set_yearDay = function(val) {
	Polycode.TimeInfo__set_yearDay(this.__ptr, val)
}

Duktape.fin(TimeInfo.prototype, function (x) {
	if (x === TimeInfo.prototype) {
		return;
	}
	Polycode.TimeInfo__delete(x.__ptr)
})
